### PR TITLE
Add subcodes error to requestAssertions

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

This PR adds subcodes error in `requestAssertions`

depends on https://github.com/kuzzleio/kuzzle/pull/1405

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :  
  ...

### CODE REVIEW

File: lib/util/esWrapper.js [hunk 1]
```diff
@@ -154,21 +154,76 @@ class ESWrapper {
...
-      case 400:
-        kuzzleError = kuzzleError || this._handleBadRequestError(error, message);
+    switch (error.displayName) {
+      case 'BadRequest':
+        if (kuzzleError) {
+          break;
+        }
+
+        if (error.body && error.body.error) {
+          message = error.body.error.root_cause
```
Comment (kuzzle) [inline lib/util/esWrapper.js]: MAJOR Correct this attempt to modify "message" or use "let" in its declaration. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3500)

File: lib/util/esWrapper.js [hunk 2]
```diff
@@ -154,21 +154,76 @@ class ESWrapper {
...
 
-      case 404:
-        kuzzleError = kuzzleError || this._handleNotFoundError(error, message);
+      case 'NotFound':
+        if (kuzzleError) {
+          break;
+        }
+
+        if (error.body && error.body.error) {
+          message = error.body.error
```
Comment (kuzzle) [inline lib/util/esWrapper.js]: MAJOR Correct this attempt to modify "message" or use "let" in its declaration. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3500)

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [esWrapper.js#L97](https://github.com/kuzzleio/kuzzle/blob/e28d34660ec33cffe35934beb05263f27ab1fc5e/lib/util/esWrapper.js#L97): Remove curly braces and "return" from this arrow function body. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3524)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [esWrapper.js#L97](https://github.com/kuzzleio/kuzzle/blob/ff06ee2ff5f3901eb3a49b63761b908d81edb59c/lib/util/esWrapper.js#L97): Remove curly braces and "return" from this arrow function body. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3524)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: doc/generate-subcodes-doc.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+      doc += `\n\n### Subdomain: ${subdomainName}, code: ${subdomain.code}\n\n`;
+      doc += '| Code | Message          | Class              | Error              | FullName           |\n------ | -----------------| ------------------ | ------------------ | ------------------ |\n';
+      for (const errorName of Object.keys(subdomain.errors)) {
+        const error = subdomain.errors[errorName];
+
+        const code = domain.code << 24
+          | subdomain.code << 16
+          | error.code;
+        doc += `\`${code}\`  | \`${error.message.replace(/%s/g, '<placeholder>')}\` | [${error.class}](https://docs.kuzzle.io/core/1/api/essentials/errors/#${error.class.toLowerCase()}) | ${errorName} | ${domainName}.${subdomainName}.${errorName}\n`;
```
Comment (Aschen) [inline doc/generate-subcodes-doc.js]:
```suggestion
        doc += `\`${code}\`  | \`${error.message.replace(/%s/g, '<placeholder>')}\` | [${error.class}](https://docs.kuzzle.io/core/2/api/essentials/errors/#${error.class.toLowerCase()}) | ${errorName} | ${domainName}.${subdomainName}.${errorName}\n`;
```
Also Kuzzle major version should be a global variable in the top of this file

File: doc/generate-subcodes-doc.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      }
+      doc += '\n---\n';
+    }
+    doc += '\n---\n';
+  }
+  const output = process.argv[2] === '-o' || process.argv[2] === '--output'
+    ? process.argv[3]
+    : './2/api/essentials/errors/subcodes/index.md';
+  fs.writeFile(output, doc, (err => {
```
Comment (Aschen) [inline doc/generate-subcodes-doc.js]: Nit: in scripts you can use `writeFileSync`, it's not necessary to have async write

File: lib/util/esWrapper.js [hunk 5]
```diff
@@ -25,7 +25,7 @@ const
   Bluebird = require('bluebird'),
```
Comment (Aschen) [inline lib/util/esWrapper.js]: Please revert all changes made to this file. You have merged the `1.x` version of this file in the `2.x` branch

**Pull request merged**